lib/resty/redis.lua

```diff
@@ -107,7 +65,7 @@ end
 
 
 function _M.set_timeout(self, timeout)
-    local sock = self.sock
+    local sock = rawget(self, "sock")
```

Comment:
This change might be a bit overkill. I'd rather stick with the much nicer dot notation :)